src/SLCore/State/ActiveConfigScopeTracker.cs

```diff
@@ -104,10 +104,17 @@ public async Task SetCurrentConfigScopeAsync(string id, string connectionId = nu
 
         using (await asyncLock.AcquireAsync())
         {
-            Debug.Assert(currentConfigScope == null);
-
-            await configurationScopeService.DidAddConfigurationScopesAsync(
-                new DidAddConfigurationScopesParams(new List<ConfigurationScopeDto> { configurationScopeDto }));
+            if (currentConfigScope != null && currentConfigScope.id == id)
```

Comment:
You can simplify if condition 
```suggestion
            if (currentConfigScope?.id == id)
```